src/expandableCalendar/calendarProvider.js

```diff
@@ -34,14 +42,86 @@ class CalendarProvider extends Component {
   };
 
   setDate = (date, updateSource) => {
-    this.setState({date, updateSource});
+    this.setState({date, updateSource}, () => {
+      this.animateTodayButton(date);
+    });
     _.invoke(this.props, 'onDateChanged', date, updateSource);
   }
+
+
+  getButtonIcon(date) {
+    const isPastDate = this.isPastDate(date);
+    return isPastDate ? require('../img/down.png') : require('../img/up.png');
+  }
+
+  isPastDate(date) {
+    const today = XDate();
+    const d = XDate(date);
+
+    if (today.getFullYear() > d.getFullYear()) {
+      return true;
+    }
+    if (today.getFullYear() === d.getFullYear()) {
+      if (today.getMonth() > d.getMonth()) {
+        return true;
+      }
+      if (today.getMonth() === d.getMonth()) {
+        if (today.getDate() > d.getDate()) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  animateTodayButton(date) {
+    if (this.props.todayButton) {
+      this.setState({buttonIcon: this.getButtonIcon(date)});
+  
+      const today = XDate().toString('yyyy-MM-dd');
+      const isToday = today === date;
+      
+      Animated.spring(this.state.buttonY, {
```

Comment:
The button disappears when picking "today" day from the calendar, but if I press the actual "Today Button" it doesn't disappear.

Reply:
Will check it out. Thanks!

Reply:
Done. Thanks!!!